### PR TITLE
Git-UI extension for Git credentials

### PR DESCRIPTION
**Pull request by chrmarti**

### CODE REVIEW

**Review by rajkumar42 — APPROVED** [review]
Thank you!

File: extensions/git-ui/src/typings/refs.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/// <reference path='../../../../src/vs/vscode.d.ts'/>
+/// <reference path='../../../../src/vs/vscode.proposed.d.ts'/>
+/// <reference types='@types/node'/>
```
Comment (joaomoreno) [inline extensions/git-ui/src/typings/refs.d.ts]: Is this still needed?

File: extensions/git-ui/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import { ExtensionContext, commands } from 'vscode';
+
+import * as cp from 'child_process';
+
+export async function deactivate(): Promise<any> {
+}
+
+export async function activate(context: ExtensionContext): Promise<void> {
+	context.subscriptions.push(commands.registerCommand('git-ui.gitCredential', async (data: any) => {
```
Comment (joaomoreno) [inline extensions/git-ui/src/main.ts]: Dashes and uppercase? How about simply `git.credential`?

File: extensions/git-ui/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+export function exec(command: string, options: cp.ExecOptions & { stdin?: string } = {}) {
+	return new Promise<ExecResult>((resolve, reject) => {
+		const child = cp.exec(command, options, (error, stdout, stderr) => {
+			(error ? reject : resolve)({ error, stdout, stderr });
+		});
+		if (options.stdin) {
+			child.stdin.write(options.stdin, (err: any) => {
+				if (err) {
+					console.error(err);
```
Comment (joaomoreno) [inline extensions/git-ui/src/main.ts]: Is there a `return` missing here?

**Review by joaomoreno — APPROVED** [review]

File: extensions/git-ui/src/typings/refs.d.ts (hunk 1 above)
Comment (chrmarti) [reply extensions/git-ui/src/typings/refs.d.ts]: It seems so. The Git extension also has its refs.d.ts. 🤷‍♂

File: extensions/git-ui/src/main.ts (hunk 3 above)
Comment (chrmarti) [reply extensions/git-ui/src/main.ts]: Not sure. I would still want to close stdin in case it is still open.

Comment (chrmarti) [reply extensions/git-ui/src/main.ts]: Actually, rejecting the promise and returning now. 👍

File: extensions/git-ui/src/typings/refs.d.ts (hunk 1 above)
Comment (joaomoreno) [reply extensions/git-ui/src/typings/refs.d.ts]: I wouldn't take the Git extension as an example, it's a dinosaur by now. 🦕

**Pull request merged**